modules/compute/gke-node-pool/main.tf

```diff
@@ -367,6 +367,22 @@ resource "google_container_node_pool" "node_pool" {
       condition     = !(var.num_node_pools < 0 || var.num_slices < 0)
       error_message = "Negative integer value of num_node_pools or num_slices is not valid. Please use a positive integer value to set num_node_pools for CPUs and GPUS, and num_slices for TPUs."
     }
+    precondition {
+      condition     = var.enable_flex_start == true ? (var.auto_repair == false) : true
+      error_message = "enable_flex_start needs node auto_repair set to false."
+    }
+    precondition {
+      condition     = var.enable_flex_start == true ? (var.static_node_count == null) : true
+      error_message = "enable_flex_start does not work with static_node_count. static_node_count should be set to null."
```

Comment:
what if it is set to 0?

Reply:
autoscaling is disabled when `static_node_count` is set to any value other than`null`.
Ref: https://github.com/GoogleCloudPlatform/cluster-toolkit/blob/80b15d3149de31114cc5b4994c56791f699a6f0f/modules/compute/gke-node-pool/variables.tf#L151